aws/scenarios/dockerVM/run.go

```diff
@@ -13,7 +13,7 @@ func Run(ctx *pulumi.Context) error {
 		return err
 	}
 
-	_, err = docker.NewAgentDockerInstaller(vm, docker.WithAgent())
+	_, err = docker.NewAgentDockerInstaller(vm.UnixVM, docker.WithAgent())
```

Comment:
`docker.NewAgentDockerInstaller` requires a `commonnVM.UnixVM`, while `ec2vm.NewUnixEc2VM(ctx)` returns `UnixEc2VM`.  `docker.NewAgentDockerInstaller` should maybe require the interface implemented by `commonnVM.UnixVM`. Is there a way to avoid this change ad let go know `UnixEc2VM` embeds a `commonnVM.UnixVM` and everything is fine ?